crates/oxc_language_server/src/worker.rs

```diff
@@ -129,12 +129,18 @@ impl WorkspaceWorker {
             || old_options.use_nested_configs() != new_options.use_nested_configs()
             || old_options.fix_kind() != new_options.fix_kind()
             || old_options.unused_disable_directives != new_options.unused_disable_directives
+            // TODO: only the TsgoLinter needs to be dropped or created
+            || old_options.type_aware != new_options.type_aware
     }
 
     pub async fn should_lint_on_run_type(&self, current_run: Run) -> bool {
-        let run_level = { self.options.lock().await.run };
+        let options = self.options.lock().await;
+        // `tsgolint` only supported the os file system. We can not run it on memory file system.
+        if options.type_aware {
+            return current_run == Run::OnSave;
```

Comment:
should we log a warning if user has configured it to lint on type, but type_aware is enabled?

Reply:
It will only be shown in VSCode in the Output Channel for OXC.
Do not know how other clients will handle it. I hope the same 🤞